src/components/Controllers/UserController.tsx

```diff
@@ -91,7 +95,29 @@ export const UserController = () => {
     };
   }, []);
 
-  return showSignModal ? (
+  useEffect(() => {
+    // Check if the address is in the sanction list
+    const checkSanctionStatus = async () => {
```

Comment:
As API has a limited usage cap, it's better to use `useQuery` and utilize its caching feature